frame/curve-amm/src/lib.rs

```diff
@@ -281,6 +279,109 @@ pub mod pallet {
 			PoolCount::<T>::get()
 		}
 
+		fn get_token_index(
+			pool_id: T::PoolId,
+			asset_id: Self::AssetId,
+		) -> Result<Self::PoolTokenIndex, DispatchError> {
+			let assets_pair = PoolAssets::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			if assets_pair.base == asset_id {
+				Ok(0.into())
+			} else if assets_pair.quote == asset_id {
+				Ok(1.into())
+			} else {
+				Err(Error::<T>::AssetNotFound.into())
+			}
+		}
+
+		fn get_price(
+			pool_id: Self::PoolId,
+			asset_id: Self::AssetId,
+			balance: Self::Balance,
+		) -> Result<Self::Balance, DispatchError> {
+			let prec = T::Precision::get();
+			let token_index = Self::get_token_index(pool_id, asset_id)?;
+
+			let pool = Self::get_pool_info(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			let assets_pair = PoolAssets::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			let assets = assets_pair.as_slice();
+			let mut balances = Vec::new();
+			for asset_id in assets {
+				balances.push(PoolAssetBalance::<T>::get(pool_id, asset_id));
+			}
+			let n_coins = assets.len();
+
+			let i = token_index.into() as usize;
+			let j = if i == 0 { 1 } else { 0 };
```

Comment:
if i == 2?

Reply:
it can rewrite so:
```
let j = i ^ 1;
```
(https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e4875ba8f5e80ac4cc7c109735b5c174)